lib/puppet/reports/servicenow.rb

```diff
@@ -8,6 +8,57 @@
   def process
     settings_hash = settings
 
+    case settings['operation_mode']
+    when 'event_management'
+      process_event_management(settings_hash)
```

Comment:
You could probably remove the call to `settings` and creating a variable on line 9 and either get the settings hash within the function and remove the parameter, or call settings on this line and pass that instead of passing a variable. Either way, settings will only get called once and this would reduce the verboseness of passing it around.

Reply:
For example you could remove line 9, and instead of creating a `settings_hash` variable at the top level of the `process` function, you could either call it as the value to pass to the two mode functions like `process_event_management(settings)`, or remove that variable from the method signature entirely and call `settings_hash = settings` as the first line of each of those two functions. Either way `settings` will still only get called once. Line 9 is no longer necessary to prevent multiple calls to the function.

Reply:
I think in a future improvement we could even make `settings` an internal method, and rest of the module code would call memoized accessors like instead of calling `settings_hash['instance']`, the module could just called `instance` and it would get the memoized value. But, anyway, that's not necessary right now. Just a thought.

Reply:
Ah ok. It still is b/c we need to get the `operation_mode` in case statement. But I can fix Line 11 to use `settings_hash`

Reply:
D'oh, haha I see what you mean.

Reply:
Done.

Reply:
re: improvement, possibly. Only reason I'd be against it is in case it conflicts w/ `Puppet::Report` object's accessors though that's probably unlikely. But yes, should be separate thing.

Reply:
Even if accessors weren't direct functions, if calling `settings` was memoized and ideally with symbols as the keys instead of strings so you could call `settings[:instance]` repeatedly, that would be an improvement.